documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java

```diff
@@ -1411,6 +1411,11 @@ public void createShortcut(String documentId, String destPath, String aclIdentit
         String sourceNodeId = currentNode.getProperty(NodeTypeConstants.EXO_SYMLINK_UUID).getString();
         currentNode = getNodeByIdentifier(session, sourceNodeId);
       }
+      // add referencable mixin to the node if isn't referencable
+      if (currentNode.isNodeType(NodeTypeConstants.NT_FOLDER ) && !currentNode.isNodeType(NodeTypeConstants.MIX_REFERENCEABLE ) && currentNode.canAddMixin(NodeTypeConstants.MIX_REFERENCEABLE)) {
```

Comment:
Why is it limited to nt:folder nodes ? can't it be applicable for other nodetypes ?

Reply:
It can  be applicable for all nodetypes, PR updated !